python/tank/__init__.py

```diff
@@ -9,6 +9,37 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 
+########################################################################
+# Establish pipeline configuration context if needed
+# 
+# When the tank command executes, or when the python API is imported
+# via the API proxy that is available with every non-localized pipeline config,
+# an environment variable TANK_CURRENT_PC, is set, pointing at the configuration
+# associated with the currently running config. This is to ensure that the right 
+# version of the code is associated with the currently running project 
+# configuration. 
+#
+# However, in the case when a cloned configuration has been localized,
+# the API proxy has been replaced by the actual core API code.
+# In this case, we will set the TANK_CURRENT_PC explicitly.
+#
+# The logic below is fundamentally to avoid the issue that when a cloned localized
+# configuration has its python sgtk/tank module imported directly, it will associate
+# itself with the primary config rather than with the config where the code is located. 
+
+import os
+if "TANK_CURRENT_PC" not in os.environ:
+    # find the pipeline configuration root, probe for a key file
```

Comment:
Hmm, the reason i didn't do this is because it is a very specific check and a sort of really special snowflake-y thing we are doing. Your comments make total sense and I am really torn. This is nothing we want to do on a regular basis. I actually initially very intentionally left it right in there, in `__init__.py`, to underline that it is a very special check. I think ultimately, we want to refactor the core API to be an atomic, pluggable thing, but we are not there yet. In the meantime, I'd rather have a change that stands out a bit more. I fear that by for example moving it into `pipelineconfig_utils`, we make it seems like a normal, standard thing - which it isn't! Hope this logic make sense.

I'll add some further comments around this to really emphasize the intent. Hope you guys feel that is sufficient!